go/ai/gen.go

```diff
@@ -203,3 +203,13 @@ type ToolResponse struct {
 	// An example might be map[string]any{"name":"Thomas Jefferson", "born":1743}.
 	Output map[string]any `json:"output,omitempty"`
 }
+
+// NewTextMessages is a helper function that creates a slice of [Message] from one role and string.
```

Comment:
I think this function is too specific. I'd rather see something like
```
func NewUserGenerateRequest(parts ...*ai.Part)
```

Reply:
None of us is exactly happy with it but the only way it's useful is if it saves the full 7-8 LOC for the simplest case, otherwise I'm not sure it's worth it. Let's discuss post-bug bash once folks have a chance to start with and extend the sample.